python/hail/linalg/matrix.py

```diff
@@ -47,27 +53,61 @@ def write(self, path):
     @handle_py4j
     @typecheck_method(that=block_matrix_type)
     def __mul__(self, that):
-        return BlockMatrix(self.hc,
-            self._jbm.multiply(that._jbm))
+        return BlockMatrix(self._jbm.multiply(that._jbm))
```

Comment:
I think I agreed with @tpoterba's comment in the meeting this should be `dot` instead of Python multiply operator to match the numpy interface.

Reply:
so this would be what---`BlockMatrix.dot(bm1, bm2)`? `linalg.dot(bm1, bm2)`? Should we preserve the __mul__? Or `bm1.dot(bm2)` (is that also valid numpy notation? I've only seen the former)

Reply:
Here's the equivalent numpy:

```
>>> A = np.random.random((20, 7))
>>> B = np.random.random((7, 30))
>>> C = A.dot(B)
>>> C.shape
(20, 30)
>>> C2 = np.dot(A, B)
>>> C2.shape
(20, 30)
```

So I vote for bm1.dot(bm2).  Let's leave out the package-level function since we haven't been doing those so far.